# NOTE: Only certain parameters are exposed here, check sample launch files for the rest
# Configuration parameters for R2D2 robot in pybullet ros simulator

# pybullet plugins, will be loaded during runtime
# -----------------------------------------------------------------------
# body_vel_control - Subscribe to cmd_vel and apply desired speed to the robot, without any noise
# odometry - Query robot base pose and speed from pybullet and publish to /odom topic
# control  - Position, velocity and effort control for all revolute joints on the robot
# joint_state_pub - Query robot state and publish position, velocity and effort values to /joint_states
# laser_scanner - Laser scanner simulation based on pybullet rayTestBatch function
plugins:
  -
    module: pybullet_ros.plugins.body_vel_control
    class: cmdVelCtrl
  -
    module: pybullet_ros.plugins.odometry
    class: simpleOdometry
  -
    module: pybullet_ros.plugins.control
    class: Control
  - 
    module: pybullet_ros.plugins.joint_state_pub
    class: joinStatePub
  - 
    module: pybullet_ros.plugins.laser_scanner
    class: laserScanner
  - 
    module: pybullet_ros.plugins.rgbd_camera
    class: RGBDCamera
  - 
    module: pybullet_ros.plugins.diff_drive
    class: DiffDrive
    name: diff_drive_controller

loop_rate:              80.0            # default: 80.0, the frequency at which to step sim in hz
gravity:                -9.81           # default: -9.81, earth gravity in m/s^2
max_effort:             100.0           # default: 50.0, the max force to apply to the model in vel mode

# True = allow pybullet to compute the inertia tensor based on mass and volume of the collision shape
# False = you can provide a more accurate inertia tensor, so inertia will be read then from URDF model
use_intertia_from_file: False           # default: False

# laser plugin params, default values are based on hokuyo URG-04LX-UG01
laser:
    frame_id:           front_laser     # default: laser, laser reference frame
    angle_min:          -1.5707963      # default: -90 deg, laser field of view (fov) min angle in rad
    angle_max:          1.5707963       # default: 90 deg, overall (with prev param) we have a 180 degree fov
    num_beams:          50              # default: 512 beams for hokuyo laser
    range_min:          0.03            # default: 0.03, the minimum distance (range) that the laser is able to detect in m
    range_max:          5.6             # default: 5.6, the maximum distance (range) that the laser is able to detect in m
    beam_visualisation: False           # default: False, draw lines on pybullet gui representing the laser beams

# rgb camera plugin parameters
rgbd_camera:
    frame_id:           box             # default: box, the name of the reference frame of the camera
    resolution:
        width:          640             # default: 640, horizontal image resolution in pixels
        height:         480             # default: 480, vertical image resolution in pixels

diff_drive_controller:
  left_joints:
    - left_wheel_joint
  right_joints:
    - right_wheel_joint
  wheel_separation: 1.0
  wheel_radius: 1.0
  cmd_vel: /cmd_vel
